### PR TITLE
GCS_MAVLink: move fixed value variable out of for

### PR DESCRIPTION
**Pull request by tajisoft**
I don't think it is necessary to put protocol_timeout inside the for statement.

### CODE REVIEW

**Comment by rmackay9** [comment]
Indeed it's not but I suspect the compiler optimises it so that it doesn't actually require the extra 4 bytes of stack.  This change also slightly increases the scope of the constant.

I'll leave this to @peterbarker but I suspect he may choose not to merge this... the change is not bad but perhaps not enough of an improvement to merge.

**Review by khancyr — APPROVED** [review]
Yep, it should be outside the loop in this case even if the compiler is smart enough to understand it

**Comment by tajisoft** [comment]
@rmackay9 @khancyr 
Thank you for your comment. I'm not familiar with compiler optimizations, but I just thought that constants should be outside the for statement from a source code point of view. It's up to you to merge.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1460,11 +1460,11 @@ GCS_MAVLINK::update_receive(uint32_t max_time_us)
 
     status.packet_rx_drop_count = 0;
 
+    const uint32_t protocol_timeout = 4000;
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp:1463]: Compiler output is identical before/after this change on Durandal.

We're moving protocol_timeout further from where it is used - it's not a stellar name for a variable in the first place, and I kind of think this makes the code marginally less readable, sorry.

**Pull request closed without merging**